app/views/directives/add-secret-to-application.html

```diff
@@ -26,18 +26,24 @@
         <legend>Add secret as:</legend>
         <div class="form-group">
           <div class="radio">
-            <label class="add-choice" for="envFrom">
-              <input id="envFrom" type="radio" ng-model="ctrl.addType" value="env" ng-disabled="ctrl.disableInputs">
-              Environment variables
-            </label>
+            <div class="add-choice">
+              <label for="envFrom">
+                <input id="envFrom" type="radio" ng-model="ctrl.addType" value="env" ng-disabled="ctrl.disableInputs">
+                Environment variables
+              </label>
+              <div class="alert alert-warning  env-warning" ng-show="ctrl.invalidEnvVars.length">
+                <span class="pficon pficon-warning-triangle-o"></span>
+                <span>Some of the keys for secret <strong>{{ctrl.secret.metadata.name}}</strong> are not valid environment variable names and will not be added.</span>.
+              </div>
+            </div>
             <div>
-              <label class="add-choice" for="mountVolume">
-                <input type="radio" ng-model="ctrl.addType" value="volume" ng-disabled="ctrl.disableInputs">
+              <label class="add-choice" for="volume">
+                <input id="volume" type="radio" ng-model="ctrl.addType" value="volume" ng-disabled="ctrl.disableInputs">
```

Comment:
If the `<label>` element wraps the `<input>`, you don't need the `for` and `id` attributes.

Reply:
If I remove it then clicking on the label does not toggle the radio. It's hard to click on just the little radio button.

Reply:
Really? Hm. It seems to work here, but maybe browser specific?

https://getbootstrap.com/docs/3.3/css/#checkboxes-and-radios

OK to leave if it's needed.

Reply:
You're right, dunno what I was doing wrong before...